### ISSUE TITLE
Bug: Stubs Misbehaving

### ISSUE THREAD

**Issue by mmk0083** [vuejs/test-utils#1905]
**Describe the bug**

Stubs appear to be broken. After some digging it appears something in Vue 3.2.45 is the culpret.

**To Reproduce**

1. Create a new vue project with `npm init vue@latest` and include `vitest`
2. Lock vue to version 3.2.44 and `npm install`
3. Add a simple component `StubMe.vue`

```vue
<template>
  <div>I should not exist</div>
</template>
```

4. Import and use your new component in `HelloWorld.vue`

```vue
<script setup lang="ts">
import StubMe from './StubMe.vue';

defineProps<{
  msg: string
}>()
</script>

<template>
  <div class="greetings">
    <h1 class="green">{{ msg }}</h1>
    <h3>
      You’ve successfully created a project with
      <a href="https://vitejs.dev/" target="_blank" rel="noopener">Vite</a> +
      <a href="https://vuejs.org/" target="_blank" rel="noopener">Vue 3</a>.
    </h3>
  </div>
  <StubMe />
</template>
```

5. Update `HelloWorld.spec.ts` to include a stub

```typescript
import { describe, it, expect } from 'vitest'

import { mount } from '@vue/test-utils'
import HelloWorld from '../HelloWorld.vue'

describe('HelloWorld', () => {
  it('works on v3.2.44', () => {
    const wrapper = mount(HelloWorld, { props: { msg: 'Hello Vitest' }, global: { stubs: { StubMe: { template: '<div>I am stubbed</div>'} } } })
    expect(wrapper.text()).toContain('I am stubbed')
  })
})
```

6. Run the tests - they should pass
7. Now change to vue 3.2.45 and re-run the tests - they will fail

**Expected behavior**

Stubs should work.

**Related information:**
- `@vue/test-utils` version: 2.x.x 
- `Vue` version: 3.2.45
- `node` version: 16.3.2
- `npm` (or `yarn`) version: 8.1.2

**Additional context**

**Comment by cexbrayat** [comment]
Hi @mmk0083 

Are you using the latest VTU version?

**Comment by mmk0083** [comment]
I have `@vue/test-utils: 2.2.6` in my project where I discovered this. I was also able to reproduce this as described above by simply starting a brand new vue project with `npm init vue@latest`

**Comment by mmk0083** [comment]
One additional point of interest - in my project I do have one set of tests with `{ stubs: { RouterLink: RouterLinkStub } }` that oddly enough is still working.

**Comment by cexbrayat** [comment]
@mmk0083 I tried to reproduce on Stackblitz, but this looks like it works?

https://stackblitz.com/edit/vitest-dev-vitest-gj33dw?file=package.json

I'm going to close the issue, but feel free to ping me with a repro on Stackblitz if you have a use case that fails and I'll take a look

**Comment by mmk0083** [comment]
@cexbrayat Thank you for looking into this. I've tried to reproduce it again myself this morning and now I can't - at least not on a clean vue project. I'm honestly not sure how I managed to break it that way yesterday so I apologize if I've wasted your time.

I do still have a failing stub in my real project, but after further troubleshooting it seems specific to trying to stub a chart from `vue-chartjs`.

Anyway - thank you again.
